buildflow/core/app/runtime/actors/pipeline_pattern/pull_process_push.py

```diff
@@ -185,7 +208,12 @@ async def process_element(element):
                 continue
             if not response.payload:
                 self._pull_percentage_counter.empty_inc()
-                await asyncio.sleep(1)
+                # TODO: test removing this
+                # await asyncio.sleep(1)
+                cpu_percent = proc.cpu_percent()
+                if cpu_percent > 0.0:
+                    # Ray doesn't like it when we try to set a metric to 0
```

Comment:
our counter type has an empty_inc method for this case

Reply:
done